rasa/core/validator.py

```diff
@@ -57,13 +58,43 @@ def verify_intents(self, ignore_warnings: bool = True) -> bool:
 
         return everything_is_alright
 
+    def verify_example_repetition_in_intents(
+        self, ignore_warnings: bool = True
+    ) -> bool:
+        """Checks if there is no duplicated example in different intents."""
+
+        everything_is_alright = True
+
+        duplication_hash = defaultdict(set)
+        duplicated_examples = []
+        for msg in self.intents.intent_examples:
+            msg_dict = msg.as_dict_nlu()
+            text = msg_dict.get("text")
+            duplication_hash[text].add(msg_dict.get("intent"))
+
+            if len(duplication_hash[text]) > 1:
```

Comment:
it looks like the first example will not be added to `duplicated_examples` and therefore will not be printed. We should print all duplicated examples, so I think this check should be done after all examples are processed